File: friends_tasks/lib/models/tarefa.dart
import 'usuario.dart';

class Tarefa {
  final int id;
  final String titulo;
  final String descricao;
  final String status;
  final Usuario dono;
  final List<Usuario> amigos;

  Tarefa({
    required this.id,
    required this.titulo,
    required this.descricao,
    required this.status,
    required this.dono,
    required this.amigos,
  });

  factory Tarefa.fromJson(Map<String, dynamic> json) => Tarefa(
        id: json['id'],
        titulo: json['titulo'],
        descricao: json['descricao'] ?? '',
        status: json['status'],
        dono: Usuario.fromJson(json['dono']),
        amigos: (json['amigos'] as List<dynamic>? ?? [])
            .map((a) => Usuario.fromJson(a))
            .toList(),
      );
}

File: api-escola-conectada/src/main/java/br/edu/escolaconectada/model/Tarefa.java
package br.edu.escolaconectada.model;

import jakarta.persistence.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "tarefas")
public class Tarefa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titulo;
    private String descricao;
    private String status; // Ex: PENDENTE, EM_ANDAMENTO, CONCLUIDA

    @ManyToOne
    @JoinColumn(name = "dono_id", nullable = false)
    private Usuario dono;

    // Relacionamento para compartilhar tarefas com amigos
    @ManyToMany
    @JoinTable(
            name = "tarefas_compartilhadas",
            joinColumns = @JoinColumn(name = "tarefa_id"),
            inverseJoinColumns = @JoinColumn(name = "usuario_id")
    )
    private Set<Usuario> amigos = new HashSet<>();

    public Tarefa() {
    }

    public Tarefa(Long id, String titulo, String descricao, String status, Usuario dono, Set<Usuario> amigos) {
        this.id = id;
        this.titulo = titulo;
        this.descricao = descricao;
        this.status = status;
        this.dono = dono;
        this.amigos = amigos;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Usuario getDono() {
        return dono;
    }

    public void setDono(Usuario dono) {
        this.dono = dono;
    }

    public Set<Usuario> getAmigos() {
        return amigos;
    }

    public void setAmigos(Set<Usuario> amigos) {
        this.amigos = amigos;
    }
}
